parser_library/src/context/code_scope.h

```diff
@@ -15,18 +15,43 @@
 #ifndef CONTEXT_CODE_SCOPE_H
 #define CONTEXT_CODE_SCOPE_H
 
+#include <variant>
+
 #include "macro.h"
 #include "utils/time.h"
+#include "variables/set_symbol.h"
 
 namespace hlasm_plugin::parser_library::context {
 class location_counter;
 class set_symbol_base;
 
+struct code_scope_variable
+{
+    set_symbol_base* ref = nullptr;
+
+    std::variant<std::monostate, set_symbol<A_t>, set_symbol<B_t>, set_symbol<C_t>> data;
```

Comment:
Maybe we could have a using for this? Something like
`using set_symbol_variant = std::variant<std::monostate, set_symbol<A_t>, set_symbol<B_t>, set_symbol<C_t>>`

Not sure where to put it though. Maybe right here?

Reply:
I am not sure if it adds anything. The type should not really be used in the rest of the code.